cli/azd/cmd/init.go

```diff
@@ -223,9 +223,25 @@ func (i *initAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 		return nil, fmt.Errorf("saving default environment: %w", err)
 	}
 
+	// If defaults exists but somehow became invalid, ask user if they want to override with current selection
+	defaultsExists, defaultsAreValid := i.accountManager.HasDefaults(ctx)
+	if !defaultsAreValid {
+		confirmMsg := "Do you want to use your current selection as defaults?"
+		confirmOverride, err := i.console.Confirm(ctx, input.ConsoleOptions{
+			Message:      confirmMsg,
+			DefaultValue: false,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("confirm for override default settings: %w", err)
+		}
+		if confirmOverride {
+			defaultsExists = !confirmOverride
```

Comment:
Might be helpful to put a little comment here as to what exactly why we do this (to draw a connection to the code on L244 that checks this and decides to save the value.